Ex4

evkingen opened the pull request:
На RxJava добавлено получение данных для RecyclerView и отображение в onComplete

evkingen committed:
применяю паттерн Repository

evkingen committed:
Реализация Observable для излучения новостной ленты

evkingen committed:
добавлен прогресс бар

evkingen committed:
Добавлена задержка 2 сек. Имитация загрузки данных.

evkingen committed:
RxJava

evkingen committed:
RxJava для получения данных для RecyclerView

colriot commented on app/src/main/java/com/alohagoha/newsapp/NewsActivity.java:
```diff
@@ -1,28 +1,47 @@
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
 
 
```
Несколько пустых строк подряд

colriot commented on app/src/main/java/com/alohagoha/newsapp/NewsActivity.java:
```diff
@@ -1,28 +1,47 @@
-                item -> this.startActivity(NewsDetailActivity.start(this, item))));
+        //recyclerView.setAdapter(new NewsDataAdapter(DataUtils.generateNews(),
+         //       item -> this.startActivity(NewsDetailActivity.start(this, item))));
+        disposable =new RepositoryImpl().getNews()
```
Форматируй код перед коммитом

colriot commented on app/src/main/java/com/alohagoha/newsapp/NewsActivity.java:
```diff
@@ -1,28 +1,47 @@
+                            .subscribe(il->recyclerView
+                                    .setAdapter(new NewsDataAdapter(il,
+                                        item->this.startActivity(NewsDetailActivity
+                                        .start(this, item)))),Throwable::printStackTrace,
```
Как ты считаешь, тут вообще можно разобраться, что происходит в нагромождении этих конструкций?))
Форматируй код, задавай понятные имена, выноси большие куски в отдельные методы (например, обновление данных в отдельный метод), стремись, чтобы код были приятно и понятно читать - потому что мы, как правило, читаем гораздо больше, чем пишем).

colriot commented on app/src/main/res/layout/activity_news_list.xml:
```diff
@@ -1,6 +1,16 @@
-    android:orientation="vertical" />
+    android:layout_height="match_parent">
+
+    <androidx.recyclerview.widget.RecyclerView xmlns:android="http://schemas.android.com/apk/res/android"
```
Пространства имён (неймспейсы) xml обычно определяют в рутовом тэге, и точно их не надо определять по нескольку раз.

colriot commented on app/src/main/res/layout/activity_news_list.xml:
```diff
@@ -1,6 +1,16 @@
+        android:id="@+id/recycler_news"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical" />
```
Нет такого атрибута у RV

colriot reviewed (changes requested):
Форматирование - всему голова. Не забывай делать это всегда (как минимум пере коммитом, пожалей читателей ;))

evkingen committed:
reformat

colriot commented on app/src/main/java/com/alohagoha/newsapp/NewsActivity.java:
```diff
@@ -14,34 +16,41 @@
         disposable.dispose();
-        disposable=null;
+        disposable = null;
+        recyclerView = null;
```
В активити это уже даже не обязательно делать, т.к после `onDestroy` активити будет собрана GC вместе со всеми её внутренностями (если ликов не посадил, например, в работе с Rx)

colriot commented on app/src/main/java/com/alohagoha/newsapp/NewsActivity.java:
```diff
@@ -14,34 +16,41 @@
     }
 
+    private void showNews(List<NewsItem> newsList) {
+        NewsDataAdapter.OnItemClickListener listener = item -> this.startActivity(NewsDetailActivity.start(this, item));
```
Пока вполне норм. В следующем задании сделаем совсем красиво, чтобы данные обновлять динамически

evkingen committed:
Прогресс бар стал поменьше
вместо e.PrintStackTrace используется Log.e андроида
вместо Thread.sleep используется .delay Rx
метод NewsDetailActivity переименован Start - > CreateIntent
